src/textual/css/_style_properties.py

```diff
@@ -808,14 +810,47 @@ def __get__(
     ) -> Color:
         """Get a ``Color``.
 
+        If the property is "color" and the styles have a "color_auto", this value will be dynamically calculated
+        by taking into account the background colors of the attached DOMNode's ancestors.
+
         Args:
             obj (Styles): The ``Styles`` object.
             objtype (type[Styles]): The ``Styles`` class.
 
         Returns:
             Color: The Color
         """
-        return cast(Color, obj.get_rule(self.name, self._default_color))
+        if self.name != "color" or not obj.color_auto:
+            # Simple (and main) case: we just return the color stored in the Style rule:
+            return cast(Color, obj.get_rule(self.name, self._default_color))
+
+        # We're dealing with a "color" property that has an "auto" value: let's traverse the Node ancestors to
+        # calculate what the resulting value should be.
+        from ..dom import DOMNode
+
+        styles_node = getattr(obj, "node", None)
+        if not isinstance(styles_node, DOMNode):
+            raise ValueError(
+                "Dynamic 'auto' color can only be used with an subclass of StylesBase that has a 'node' attribute"
+            )
+        # We just repeat what `DOMNode.color` does, in a slightly simplified way.
```

Comment:
Have you considered rolling this logic in to the `color` property itself?

Reply:
The thing is that `DOMNode.colors` returns 4 computed colours - foreground and background colours, and its parent's colours. Whereas the Style properties each manage their single property